test/runtests.jl

```diff
@@ -263,6 +263,23 @@ end
     end
 end
 
+@testset "heap-to-stack optimization" begin
```

Comment:
Maybe we want to do these checks within the existing test cases, rather than separating like this.

Reply:
Good point!